prototype/galore/galore_fused/utils.py

```diff
@@ -0,0 +1,111 @@
+import torch
+
+
+def get_orthogonal_matrix(weights, rank, type):
+    module_params = weights
+
+    if module_params.data.dtype != torch.float:
+        float_data = False
+        original_type = module_params.data.dtype
+        original_device = module_params.data.device
+        matrix = module_params.data.float()
+    else:
+        float_data = True
+        matrix = module_params.data
+
+    U, s, Vh = torch.linalg.svd(matrix, full_matrices=False)
+
+    # make the smaller matrix always to be orthogonal matrix
+    if type == "right":
+        A = U[:, :rank] @ torch.diag(s[:rank])
```

Comment:
This is unused; there's a PR upstream: jiaweizzhao/GaLore#18
Same comment on line 28 as well.